src/Alpacka.Lib/Curse/ResourceHandlerCurse.cs

```diff
@@ -52,60 +52,62 @@ public class ResourceHandlerCurse : IResourceHandler
-            
-            foreach (var dep in fileInfo.Dependencies) {
-                if (dep.Type == DependencyType.Required) {
-                    var depAddon = await CurseMeta.GetAddon(dep.AddonId);
-                    var depMod = new EntryMod {
-                        Name    = depAddon.Name,
-                        Handler = Name,
-                        Source  = dep.AddonId.ToString(),
-                        Version = Release.Latest.ToString(), // avoid crashes from listing files
-                        Side    = mod.Side,
-                    };
-                    _modToDependencyType[depMod] = dep.Type;
-                    addDependency(depMod);
-                } else if (dep.Type == DependencyType.Optional) {
-                    var depAddon = await CurseMeta.GetAddon(dep.AddonId);
-                    // TODO: Make this available in some form in the return value.
-                    Console.WriteLine($"'{ mod.Name }' recommends using '{ depAddon.Name }'");
+            using (var curseMeta = new CurseMeta())
+            {
+                var addon = await curseMeta.GetAddon(id);
+                
+                mod.Name        = mod.Name ?? addon.Name;
+                mod.Description = mod.Description ?? addon.Summary;
+                if (mod.Links == null) mod.Links = new EntryLinks();
+                mod.Links.Website   = mod.Links.Website ?? addon.WebSiteURL;
+                mod.Links.Source    = mod.Links.Source ?? addon.ExternalUrl;
+                mod.Links.Donations = mod.Links.Donations ?? addon.DonationUrl;
+                
+                
+                var fileId = await FindFileId(curseMeta, addon, mod, mcVersion, optional);
+                if (fileId == -1) {
+                    if (optional) {
+                        Debug.WriteLine($"no file found for { mod.Source } This is not a Error");
+                        return null; // We do not throw a error because its not required
+                    // We should probably not reach this point ever:
+                    } else throw new Exception($"No File of type 'Release' found for { mod.Name } in { mcVersion }");
                 }
+                
+                var fileInfo = await curseMeta.GetAddonFile(addon.Id, fileId);
+                mod.Source = fileInfo.DownloadURL;
+                mod.Path  = Path.Combine(mod.Path, fileInfo.FileNameOnDisk);
+                
+                foreach (var dep in fileInfo.Dependencies) {
+                    if (dep.Type == DependencyType.Required) {
+                        var depAddon = await curseMeta.GetAddon(dep.AddonId);
+                        var depMod = new EntryMod {
+                            Name    = depAddon.Name,
+                            Handler = Name,
+                            Source  = dep.AddonId.ToString(),
+                            Version = Release.Latest.ToString(), // avoid crashes from listing files
```

Comment:
it does no longer apply.. actually this refers to a behaviour that was removed when i moved the curse api stuff into alpacka-meta, it used to either "just" get the recommended file
or get all files and look for the best one to pick

now it always gets info for all files per mod since its the same effort, it includes all deleted files we know about.. and its is safe